### PR TITLE
Bug fix to filter out rules and amendments based on active flag

### PR DESCRIPTION
**Pull request by Kirbitz**
Ben caught a bug in the original PR where the data was not being filtered out based on the active flag this PR is for that bug

- RulesAmendmentsRepository updated to filter rules and amendments based on the active flag
- Tests were added to check for proper filtering
- Added active attribute to RuleDTO
- Removed Active attribute from the AmendmentsModel
- Added new test data for the tests

### CODE REVIEW

File: src/test/kotlin/nomic/api/RulesAmendmentsTests.kt [hunk 1]
```diff
@@ -44,6 +51,14 @@ class RulesAmendmentsTests(@Autowired val client: TestRestTemplate) {
         Assertions.assertThat(result?.size).isGreaterThan(1)
     }
 
+    @Test
+    fun `Found Game With Rules and Amendment Activity Equals False`() {
+        val entity = client.getForEntity<List<LinkedHashMap<String, List<AmendmentModel>>>>("/api/rules_amendments/1")
+        val result = entity.body?.get(3)?.get("amendments")
+        Assertions.assertThat(entity.statusCode).isEqualTo(HttpStatus.OK)
+        Assertions.assertThat(result?.size).isEqualTo(0)
```
[the remark this answers is not in the document]
Comment (Kirbitz) [reply src/test/kotlin/nomic/api/RulesAmendmentsTests.kt:59]: The system is just really weird about deserializing the json data it just kept complaining about formatting errors.  If you want to take a stab at it please be my guest, but this was the best I could figure out.
To break it down the json is a list then the object is read as a hash map with strings as the key value then finally I can dig down to the AmendmentModel to check for number of existing items

**Review by meyert8034 — APPROVED** [review]
Changes look good, all tests still pass.

File: src/test/resources/data.sql [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+INSERT INTO `Rule` VALUES (3, 3, 'Jump off a bridge', 'MyRule3', 1, 1, 0);
+INSERT INTO `Amendment` VALUES (2, 2, 'Take a photo while doing it', 'MyAmendment2', 3, 1);
+INSERT INTO `Amendment` VALUES (3, 3, 'Smile real wide', 'MyAmendment3', 3, 1);
+INSERT INTO `Rule` VALUES (5, 5, 'Talk politics', 'MyRule4', 1, 1, 0);
+INSERT INTO `Amendment` VALUES (5, 5, 'Also talk religion', 'MyInactiveAmendment1', 5, 0);
+
+INSERT INTO `Game` VALUES (2, 'MyGame2', '2015-12-18', NULL, 2);
+
+INSERT INTO `Game` VALUES (3, 'MyGame3', '2015-12-18', NULL, 2);
+INSERT INTO `Rule` VALUES (4, 4, 'Run in circles', 'MyInactiveRule1', 3, 0, 0);
```
[the remark this answers is not in the document]
Comment (Kirbitz) [reply src/test/resources/data.sql:18]: Nah it is all just stuff I came up with on the spot lol.  I like to have fun with the test cases

**Review by AdamLucka — APPROVED** [review]
Everything looks good, the bug of not ignoring rules and amendments with the active flag set to false has been fixed. There are tests present that verify the fix was successful. Also, some general refactoring was done to simplify exception handling in the future.

**Review by meyert8034 — APPROVED** [review]
Good bugfix, changes to areas commented on looks good.

**Review by flare199 — APPROVED** [review]
Everything ran and works as intended. No issues or problems found. Good work Meralfkie keep it up! :) <3

**Pull request merged**